src/plugins/mer/merbuildconfiguration.cpp

```diff
@@ -82,6 +100,31 @@ void MerBuildConfiguration::initialize(const ProjectExplorer::BuildInfo *info)
     ensureHasVmStartStep();
 }
 
+QList<NamedWidget *> MerBuildConfiguration::createSubConfigWidgets()
+{
+    auto subConfigWidgets = QmakeBuildConfiguration::createSubConfigWidgets();
+    auto buildSettingsWidget = new MerBuildSettingsWidget(m_useExtraParserArguments);
+    subConfigWidgets.prepend(buildSettingsWidget);
+    connect(buildSettingsWidget, &MerBuildSettingsWidget::importVariablesAutomaticallyChanged,
+            this, &MerBuildConfiguration::setUseExtraParserArguments);
+    return subConfigWidgets;
+}
+
+QVariantMap MerBuildConfiguration::toMap() const
+{
+    QVariantMap map(QmakeBuildConfiguration::toMap());
+    map.insert(MER_USE_EXTRA_VARIABLES_KEY, m_useExtraParserArguments);
+    return map;
+}
+
+bool MerBuildConfiguration::fromMap(const QVariantMap &map)
+{
+    if (!QmakeBuildConfiguration::fromMap(map))
+        return false;
+    setUseExtraParserArguments(map.value(MER_USE_EXTRA_VARIABLES_KEY).toBool());
```

Comment:
Maybe we should suppress calling `setupExtraParserArguments()` at this point and let it use what it stored last time.

Reply:
That is exactly what happens here: `setUseExtraParserArguments` assign stored value to `m_useExtraParserArguments`, then call `setupExtraParserArguments`. The execution way of `setupExtraParserArguments` internally depends on `m_useExtraParserArguments` value.

Reply:
I mean the `QMakeStep` takes itself care of (re)storing its `extraParserArguments`, we do not need to call `setupExtraParserArguments` at this point (as it is done indirectly by `setUseExtraParserArguments`). If we do not need to, maybe we should not do it. So what I suggest is to consider setting `m_useExtraParserArguments` directly here instead of using `setUseExtraParserArguments`.

```
m_useExtraParserArguments = map.value(MER_USE_EXTRA_VARIABLES_KEY).toBool()
```

Reply:
> QMakeStep takes itself care of (re)storing its extraParserArguments

It is not implemented right now. Should I implement it in `QMakeStep::toMap()`, `QMakeStep::fromMap()`?
Please, note that `m_extraArgs` not stored too, while `m_userArgs` are stored. IMHO, we should not store `m_extraParserArgs`, just like `m_extraArgs`.

Reply:
ah, lgtm then.